src/serialization/KiotaSerializer.Deserialization.cs

```diff
@@ -21,29 +23,34 @@ public static partial class KiotaSerializer
     /// <param name="contentType">The content type of the stream.</param>
     /// <param name="parsableFactory">The factory to create the object.</param>
     /// <param name="serializedRepresentation">The serialized representation of the object.</param>
+    //[Obsolete("Use DeserializeAsync instead")]
     public static T? Deserialize<T>(string contentType, string serializedRepresentation, ParsableFactory<T> parsableFactory) where T : IParsable
     {
         if(string.IsNullOrEmpty(serializedRepresentation)) throw new ArgumentNullException(nameof(serializedRepresentation));
         using var stream = GetStreamFromString(serializedRepresentation);
+#pragma warning disable CS0618 // Type or member is obsolete
```

Comment:
I'd avoid obsolete suppressions as it makes for inconsistent APIs, but let's come to an agreement on the issue before we make any further obsolete changes